File: harvester-bot.js
const CONSTANTS = require('./helper-package-js/constants');
const { ACTION, FACE, FLOWER, MOVE } = require('./helper-package-js/constants');

const { Action } = require('./helper-package-js/action');
const { Bot } = require('./helper-package-js/bot');
const { Position } = require('./helper-package-js/position');

class HarvesterBot extends Bot {
    constructor() {
        super("HarvesterBot");
    }

    getActions() {
        const actions = [];

        for (let beeCell of this.beeCells) {
            const bee = beeCell.bee;
            const pos = beeCell.pos;
            // If the bees have pollen, move to queen.
            if (bee.pollen >= bee.count) {
                if (!this.isBesideHiveOrQueen(pos)) {
                    const path = this.getMinPath(pos, this.queenBee.pos);
                    actions.push(new Action({
                        type: ACTION.MOVE,
                        pos: pos,
                        move: path.move,
                        face: path.move - 1 /* Faces move direction. */,
                    }));
                }
            } else {
                // Move to closest flower to collect pollen.
                const firstFlower = this.flowerCells[0];
                let minPath = this.getMinPath(pos, firstFlower.pos);
                for (let i = 1; i < this.flowerCells.length; ++i) {
                    const path = this.getMinPath(pos, this.flowerCells[i].pos);
                    if (path.distance < minPath.distance) {
                        minPath = path;
                    }
                }
                actions.push(new Action({
                    type: ACTION.MOVE,
                    pos: pos,
                    move: minPath.move,
                    face: minPath.move - 1 /* Faces move direction. */,
                }));
            }
        }

        return actions;
    }
}

(async function main() {
    const harvesterBot = new HarvesterBot();
    await harvesterBot.run();
})();

File: harvester_bot.py
from helper_package_python.constants import ActionType, Face

from helper_package_python.bot import Bot
from helper_package_python.action import Action

class HarvesterBot(Bot):
	def __init__(self):
		super().__init__("HarvesterBot")

	def init(self):
		pass

	def getActions(self):
		actions = []

		for beeCell in self.beeCells:
			bee = beeCell.bee
			pos = beeCell.pos

			# If the bees have pollen, move to queen.
			if bee.pollen >= bee.count:
				if not self.isBesideHiveOrQueen(pos):
					path = self.getMinPath(pos, self.queenBee.pos)

					actions.append(Action(
						ActionType.MOVE,
						pos,
						path.move,
						Face(path.move - 1) # Face move direction
					))
			else:
				# Move to closest flower to collect pollen.
				firstFlower = self.flowerCells[0]
				minPath = self.getMinPath(pos, firstFlower.pos)

				for i in range(1, len(self.flowerCells)):
					path = self.getMinPath(pos, self.flowerCells[i].pos)
					if path.distance < minPath.distance:
						minPath = path

				actions.append(Action(
					ActionType.MOVE,
					pos,
					minPath.move,
					Face(minPath.move - 1) # Face move direction
				))

		return actions

def main():
	harvesterBot = HarvesterBot()
	harvesterBot.run()

main()
